example-gto/generate.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+
+# Setup script env:
+#   e   Exit immediately if a command exits with a non-zero exit status.
+#   u   Treat unset variables as an error when substituting.
+#   x   Print commands and their arguments as they are executed.
+set -eux
+
+PUSH=false
+echo $#
+if [ "$#" -eq 1 ] && [ "$0" != "--push" ]; then
+   PUSH=true;
+   echo "Will push things to GitHub :tada:"
+fi
+
+HERE="$( cd "$(dirname "$0")" ; pwd -P )"
+USER_NAME="iterative"
+REPO_NAME="example-gto"
+
+BUILD_PATH="$HERE/build"
+
+mkdir $BUILD_PATH
+pushd $BUILD_PATH
+if [ ! -d "$BUILD_PATH/.venv" ]; then
+  virtualenv -p python3 .venv
+  export VIRTUAL_ENV_DISABLE_PROMPT=true
+  source .venv/bin/activate
+  echo '.venv/' > .gitignore
+  pip install gitpython
+  pip install gto
+fi
+popd
+
+source $BUILD_PATH/.venv/bin/activate
+
+REPO_PATH="$HERE/build/$REPO_NAME"
+
+if [ -d "$REPO_PATH" ]; then
```

Comment:
this should probably go before ... I would keep e2e reproducible, so that builds do not affect each other in any way

Reply:
Thanks for the comment, made changes.